packages/autocomplete-js/src/__tests__/autocomplete.test.ts

```diff
@@ -151,6 +155,90 @@ describe('autocomplete-js', () => {
     `);
   });
 
+  test("renders with an auto-incremented id if there's multiple instances", async () => {
+    const container = document.createElement('div');
+
+    document.body.appendChild(container);
+    autocomplete({
+      container,
+    });
+    autocomplete({
+      container,
+    });
+    autocomplete({
+      container,
+    });
+
+    await waitFor(() => {
+      expect(
+        document.querySelector('#autocomplete-0-label')
+      ).toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-1-label')
+      ).toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-2-label')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test("doesn't increment the id when toggling detached mode", async () => {
+    const container = document.createElement('div');
+
+    document.body.appendChild(container);
+    const { update } = autocomplete<{ label: string }>({
+      container,
+    });
+
+    await waitFor(() => {
+      expect(
+        document.querySelector('.aa-DetachedSearchButton')
+      ).not.toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-3-label')
+      ).toBeInTheDocument();
+    });
+
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn((query) => ({
+        matches: true,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+
+    update({ detachedMediaQuery: '' });
+
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn((query) => ({
+        matches: false,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
```

Comment:
related: [jest doc](https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom), [matchMedia](https://github.com/algolia/autocomplete/blob/next/scripts/jest/setupTests.ts#L7-L19)

I had to overwrite it because our default `query` isn't sent to `matchMedia`, so I couldn't add a condition here. Then I had to set it back to the default implementation to make the rest of the suite works.

I'll try to play around with it but any help will help :)